.github/workflows/reusable_docker_pipeline.yml

```diff
@@ -0,0 +1,77 @@
+name: Docker Build & Publish
+
+on:
+  workflow_call:
+    inputs:
+      dockerfile:
+        required: false
+        type: string
+        description: "Path to Dockerfile"
+        default: "Dockerfile"
+      dockerContext:
+        required: false
+        type: string
+        description: "The Docker context"
+        default: "."
+      publish:
+        required: true
+        type: boolean
+
+jobs:
+  docker_build:
+    runs-on: ubuntu-22.04
```

Comment:
Is there latest ubuntu on GH Actions?

Reply:
`ubuntu-latest` is `ubuntu-22.04`. Wondering should we use `ubuntu-latest`?

Reply:
Nope, always better to pin to specific version. Just surprised they don't have ubuntu 24 - CircleCI has already.